src/model/schema.js

```diff
@@ -465,7 +465,7 @@ export class MarkType extends SchemaItem {
 // with node type constructors and another similar object associating
 // mark names with mark type constructors.
 export class SchemaSpec {
-  // :: (?Object<NodeType>, ?Object<MarkType>)
+  // :: (?Object<SubClass<NodeType>>, ?Object<SubClass<MarkType>>)
```

Comment:
I'm also making use of static properties on these constructors in the schema-building code. Since classes/constructors are something that already exists in the language, I think it's nice to be able to use that rather than create a new concept that covers the same uses.